src/Configuration/Configuration.php

```diff
@@ -47,18 +47,35 @@ public function set(string $name, $value, callable $callback = null): void
         if (is_callable($callback)) {
             $this->validations[$name] = $callback;
         }
-        if (array_key_exists($name, $this->validations) 
-            && !call_user_func($this->validations[$name], $value)
-        ) {
-            throw new ConfigurationException("Config option \"$name\" has an invalid value.");
-        }
         $callingFile = pathinfo(debug_backtrace()[1]['file'], PATHINFO_FILENAME);
-        if (!$this->has($name) && $callingFile === 'deploy') {
-            $this->verifyConfigurationName($name);
+        if ($callingFile === 'deploy') {
```

Comment:
What if recipe called something else?  This should be a general solution. I can merge it with hard coded recipe name.

Reply:
Yes, sure, sorry! Can I match it against `$GLOBALS['deployFile']` or there is a better way? 

Thanks for you patience!

Reply:
This is also a bad move. Just check if it second set via levingshtain.

Reply:
With just the levenshtein, it could trigger the warning for correct properties as `bin/git` after `bin/dir` being applied, as they are close enough. That's why I had included the `PREDEFINED_PARAMS`. It was working as a whitelist for some of the correct set names.

I couldn't think a better way to make it work by the moment other than using the `PREDEFINED_PARAMS` whitelist or checking if the set came from the user configuration file.